src/panoptes/pocs/focuser/birger.py

```diff
@@ -249,36 +253,32 @@ def _send_command(self, command, response_length=None):
         # Clear the input buffer in case there's anything left over in there.
         self._serial_port.reset_input_buffer()
 
-        # Send command
-        self._serial_io.write(command + '\r')
-
-        # In verbose mode adaptor will first echo the command
-        echo = self._serial_io.readline().rstrip()
-        assert echo == command, self.logger.warning("echo != command: {} != {}".format(
-            echo, command))
-
-        # Adaptor should then send 'OK', even if there was an error.
-        ok = self._serial_io.readline().rstrip()
-        assert ok == 'OK'
-
         # Depending on which command was sent there may or may not be any further
         # response.
         response = []
 
-        if response_length == 0:
-            # Not expecting any further response. Should check the buffer anyway in case an error
-            # message has been sent.
-            if self._serial_port.in_waiting:
-                response.append(self._serial_io.readline())
+        for i in range(self._max_command_attempts):
+            # Send the command
+            self._serial_io.write(command + '\r')
 
-        elif response_length > 0:
-            # Expecting some number of lines of response. Attempt to read that many lines.
-            for i in range(response_length):
-                response.append(self._serial_io.readline())
+            # In verbose mode adaptor will first echo the command
+            echo = self._serial_io.readline().rstrip()
 
-        else:
-            # Don't know what to expect. Call readlines() to get whatever is there.
-            response.append(self._serial_io.readlines())
+            # Adaptor should then send 'OK', even if there was an error.
+            ok = self._serial_io.readline().rstrip()
+
+            if echo == command and ok == 'OK':
+                # Get any avaialble response left in buffer.
+                response.append(self._serial_io.readlines())
+
+                self.logger.debug(f"Got correct response after {i + 1} attempts")
```

Comment:
It's debatable but I feel that this message is unnecessary, or at least should be downgraded from DEBUG to TRACE level. If retries are needed then they will be logged and this message won't add much info, and if it works first time like it's supposed to we probably don't need a additional debug message either.